File: Java/1_Array/LC_2682_FindTheLosersOfTheCircularGame.java
import java.util.*;

class Solution_2682 {
    public int[] circularGameLosers(int n, int k) {
        boolean[] visited = new boolean[n];

        int jump = k;
        int index = 0;
        while (!visited[index]) {
            visited[index] = true;
            index = (index + jump) % n;
            jump += k;
        }

        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                list.add(i + 1);
            }
        }

        int[] ans = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            ans[i] = list.get(i);
        }

        return ans;
    }
}

public class LC_2682_FindTheLosersOfTheCircularGame {

}

File: JavaScript/1_Array/LC_2682_FindTheLosersOfTheCircularGame.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number[]}
 */
var circularGameLosers = function (n, k) {
    var visited = []
    var index = 0;
    var jump = k;
    while (!visited[index]) {
        visited[index] = true;
        index = (index + jump) % n;
        jump = jump + k;
    }

    var res = [];
    var j = 0;

    for (i = 0; i < n; i++) {
        if (!visited[i]) {
            res[j++] = i + 1;
        }
    }

    return res;
};